front-end/src/app/home/components/dashboard/dashboard.component.ts

```diff
@@ -1,20 +1,65 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
 import { CookieService } from 'ngx-cookie-service';
 import { UserService } from 'src/app/services/user.service';
+import { YahooService } from 'src/app/services/yahoo.service';
+
 @Component({
   selector: 'app-dashboard',
   templateUrl: './dashboard.component.html',
   styleUrls: ['./dashboard.component.scss'],
 })
-export class DashboardComponent {
+export class DashboardComponent implements OnInit {
+  stocks?: string[];
+  hasNoStocks?: boolean;
+  selectedStock?: string;
+  stockObject?: any;
+
   constructor(
     private cookieService: CookieService,
-    private _userService: UserService
+    private _userService: UserService,
+    private _yahooService: YahooService,
+    private router: Router,
   ) {}
 
-  deconnexion() {
-    //TODO: https://github.com/levydanqc/stockaccino/issues/7
-    this.cookieService.delete('id');
-    location.reload();
+  selectStock(stock?: string, i?: number) {
+    if (i) {
+      if (this.stocks)
+        this.selectedStock = this.stocks[i];
+    }
+    else if (stock)
+      this.selectedStock = stock;
+    
+    if (this.selectedStock) {
+      this._yahooService.getSearchedStock(this.selectedStock).subscribe((data: any) => {
+        this.stockObject = data['quoteResponse']['result'][0];
+      });
+    }
+  }
+
+  voirStock() {
+    this.router.navigate(['/search'], {
+      queryParams: { searchedStock: this.selectedStock },
+    });
+  }
+  
+  unwatchStock(stock?: string) {
```

Comment:
not used ?

Reply:
@OlivierPoitras why is this uncommented ? If not used maybe it can be deleted

Reply:
outdated